### PR TITLE
[Doc] add a section demonstrating the `<Datagrid>` headers pinning feature 

### PR DESCRIPTION
**Pull request by adguernier**
## Problem
`<Datagrid>` support headers pinning by default but it is not mentioned anywhere.

## Solution

Mention headers pinning support in the <Datagrid> documentation and add a screencast

### CODE REVIEW

**Review by slax57 — APPROVED** [review]

File: docs/Datagrid.md [hunk 1]
```diff
@@ -832,6 +832,15 @@ const PostList = () => (
 
 **Tip**: The `Datagrid` component `classes` can also be customized for all instances of the component with its global css name `"RaDatagrid"` as [describe here](https://marmelab.com/blog/2019/12/18/react-admin-3-1.html#theme-overrides)
 
+## Headers pinning
```
Comment (fzaninotto) [inline docs/Datagrid.md:835]:
```suggestion
## Header Pinning
```

File: docs/Datagrid.md [hunk 2]
```diff
@@ -832,6 +832,15 @@ const PostList = () => (
...
+
+`<Datagrid>` pins headers by default and you don't need to do anything extra for it to work:
```
Comment (fzaninotto) [inline docs/Datagrid.md:837]:
```suggestion
`<Datagrid>` has sticky headers, which means that the header row will remain visible even when scrolling down the page.
```

Comment (fzaninotto) [reply docs/Datagrid.md:837]: Here I'm using several names for the same feature (sticky headers, header pinning) to make it more discoverable

File: docs/Datagrid.md [hunk 3]
```diff
@@ -832,6 +832,15 @@ const PostList = () => (
...
+
+<video controls autoplay playsinline muted loop>
+  <source src="./img/datagrid-headers-pinning.mp4" type="video/mp4"/>
+  Your browser does not support the video tag.
+</video>
+
```
Comment (fzaninotto) [inline docs/Datagrid.md:843]:
```suggestion

You don't need to do anything for this to work, as it's enabled by default. 
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/Datagrid.md [hunk 4]
```diff
@@ -832,15 +832,16 @@ const PostList = () => (
...
 
-`<Datagrid>` pins headers by default and you don't need to do anything extra for it to work:
+`<Datagrid>` has sticky headers, which means that the header row will remain visible even when scrolling down the page.
 
 <video controls autoplay playsinline muted loop>
   <source src="./img/datagrid-headers-pinning.mp4" type="video/mp4"/>
   Your browser does not support the video tag.
 </video>
 
+You don't need to do anything for this to work, as it's enabled by default.
```
Comment (djhi) [inline docs/Datagrid.md:844]:
```suggestion
You don't need to do anything for this to work, as it's enabled by default. 

```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

File: docs/Datagrid.md (hunk 4 above)
Comment (adguernier) [reply docs/Datagrid.md:844]: I don't understand this suggestion because it seems to do the same thing as before :thinking:

**Review by djhi — APPROVED** [review]

**Pull request merged**